common/utils/src/main/scala/org/apache/spark/SparkThrowableHelper.scala

```diff
@@ -74,6 +74,26 @@ private[spark] object SparkThrowableHelper {
     errorClass.startsWith("INTERNAL_ERROR")
   }
 
+  def isRuntimeUserError(e: Throwable): Boolean = {
```

Comment:
Would it make sense that this as an optional field to error-classes.json?

Reply:
Yea I can add an optional bool flag.

I need some suggestions on the naming. We want to avoid the "job aborted" wrapper for user-facing errors, and user-facing errors are errors with error classes. I think "user-facing" should be a good name. Another case is task retry, I picked the name "user error", which may not be proper as what we really care is if the error is runtime and non-transient.

Reply:
also cc @superdupershant

Reply:
Aren't retriable/transient errors in the minority? Shouldn't we flag those instead?

Reply:
Ideally I agree with you, but the current behavior is to retry all errors, and I'm afraid of introducing regressions by not retrying most of the errors, as it's hard to identify all the errors that need retry.